src/config/mod.rs

```diff
@@ -0,0 +1,129 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module handles all configuration of this library.
+
+use blockchain::types::address::Address;
+use std::env;
+
+// Environment variables and their defaults
+const ENV_ORIGIN_ENDPOINT: &str = "MOSAIC_ORIGIN_ADDRESS";
+const DEFAULT_ORIGIN_ENDPOINT: &str = "http://127.0.0.1:8545";
+const ENV_AUXILIARY_ENDPOINT: &str = "MOSAIC_AUXILIARY_ADDRESS";
+const DEFAULT_AUXILIARY_ENDPOINT: &str = "http://127.0.0.1:8546";
+const ENV_ORIGIN_CORE_ADDRESS: &str = "MOSAIC_ORIGIN_CORE_ADDRESS";
+
+/// Global config for running a mosaic node.
+#[derive(Default)]
+pub struct Config {
+    /// Address of the origin chain, e.g. "127.0.0.1:8485"
+    origin_endpoint: String,
+    /// Address of the auxiliary chain, e.g. "127.0.0.1:8486"
+    _auxiliary_endpoint: String,
+    /// The address of a core address on origin.
+    /// It is optional as it may not be needed depending on the mode that the node is run in.
+    _origin_core_address: Option<Address>,
+}
+
+impl Config {
+    /// Reads the configuration from environment variables and creates a new Config from them. In
+    /// case an environment variable is not set, a default fallback will be used.
+    pub fn new() -> Config {
+        let origin_endpoint =
+            Self::read_environment_variable(ENV_ORIGIN_ENDPOINT, Some(DEFAULT_ORIGIN_ENDPOINT));
+        let auxiliary_endpoint = Self::read_environment_variable(
+            ENV_AUXILIARY_ENDPOINT,
+            Some(DEFAULT_AUXILIARY_ENDPOINT),
+        );
+
+        let origin_core_address = Self::read_environment_variable(ENV_ORIGIN_CORE_ADDRESS, None);
+        let origin_core_address = match origin_core_address {
+            // TODO: Can this be done in a  better way than juggling a ton of Result/Option?
+            Some(origin_core_address) => Some(Address::from_string(&origin_core_address).unwrap()),
+            None => None,
+        };
+
+        Config {
+            origin_endpoint: match origin_endpoint {
+                Some(origin_endpoint) => origin_endpoint,
+                None => panic!("An origin endpoint must be set!"),
+            },
+            _auxiliary_endpoint: match auxiliary_endpoint {
+                Some(auxiliary_endpoint) => auxiliary_endpoint,
+                None => panic!("An auxiliary endpoint must be set!"),
+            },
+            _origin_core_address: origin_core_address,
+        }
+    }
+
+    fn read_environment_variable(name: &str, default_value: Option<&str>) -> Option<String> {
+        let value = env::var(name);
+        let value = match value {
```

Comment:
what should be our best practice / style guide wrt such redeclaration; I would prefer
```
let value = match env::var(name) {
```

Reply:
Agreed 👍